code/modules/antagonists/voidwalker/voidwalker_void_eater.dm

```diff
@@ -0,0 +1,89 @@
+/**
+ * An armblade that pops windows
+ */
+/obj/item/void_eater
+	name = "void eater" //as opposed to full eater
+	icon = 'icons/obj/weapons/voidwalker_items.dmi'
+	icon_state = "tentacle"
+	inhand_icon_state = "tentacle"
+	force = 25
+	armour_penetration = 35
+	lefthand_file = 'icons/mob/inhands/antag/voidwalker_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/antag/voidwalker_righthand.dmi'
+	blocks_emissive = EMISSIVE_BLOCK_NONE
+	item_flags = ABSTRACT | DROPDEL
+	resistance_flags = INDESTRUCTIBLE | ACID_PROOF | FIRE_PROOF | LAVA_PROOF | UNACIDABLE
+	w_class = WEIGHT_CLASS_HUGE
+	sharpness = SHARP_EDGED
+	tool_behaviour = TOOL_MINING
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	wound_bonus = -30
+	bare_wound_bonus = 20
+
+/obj/item/void_eater/Initialize(mapload)
+	. = ..()
+	ADD_TRAIT(src, TRAIT_NODROP, HAND_REPLACEMENT_TRAIT)
+	AddComponent( \
+		/datum/component/butchering, \
+		speed = 8 SECONDS, \
+		effectiveness = 70, \
+	)
+
+	AddComponent(/datum/component/temporary_glass_shatterer)
+
+/obj/item/void_eater/attack(mob/living/target_mob, mob/living/user, params)
+	if(!ishuman(target_mob))
+		return ..()
+
+	var/mob/living/carbon/human/hewmon = target_mob
+
+	if(hewmon.has_trauma_type(/datum/brain_trauma/voided))
+		// explode into glass wooooohhoooo
+		var/static/list/shards = list(/obj/item/shard = 2, /obj/item/shard/plasma = 1, /obj/item/shard/titanium = 1, /obj/item/shard/plastitanium = 1)
+		for(var/i in 1 to rand(4, 6))
+			var/shard_type = pick_weight(shards)
+			var/obj/shard = new shard_type (get_turf(hewmon))
+			shard.pixel_x = rand(-16, 16)
+			shard.pixel_y = rand(-16, 16)
```

Comment:
If you're copy/pasting this just make a random shard spawner effect like gibspawners